server/packet_handlers.py

```diff
@@ -229,59 +232,105 @@ async def ping_handler(session: "Session", packet_data: bytes):
     pass
 
 
+# TODO:TEMPFIX: make this stateless
+who_spectating_who = {}
+
 # START_SPECTATING = 16
 
+
 @bancho_handler(packets.ClientPackets.START_SPECTATING)
 async def start_spectating_handler(session: "Session", packet_data: bytes):
     assert session["presence"] is not None
-    
+
     packet_reader = packets.PacketReader(packet_data)
     host_account_id = packet_reader.read_i32()
 
     host_session = await sessions.fetch_by_account_id(host_account_id)
-    
+
     if host_session is None:
         logger.warning(
             "A user attempted to spectate another user who is offline",
             spectator_id=session["account_id"],
-            host_id=host_session["account_id"],
+            host_id=host_account_id,
         )
         return
 
-    host_account_UUID = await spectators.start_spectating(
-        host_account_id,
+    await spectators.add(
         host_session["session_id"],
+        session["session_id"],
     )
-    
+
+    # TODO: make this stateless
+    who_spectating_who[session["session_id"]] = host_session["session_id"]
+
     await packet_bundles.enqueue(
-        host_account_UUID,
-        packets.write_spectator_joined_packet(session["account_id"])
+        host_session["session_id"],
```

Comment:
packets are sent to sessions, not accounts. one account may have multiple sessions (e.g. tournament client)